### ISSUE TITLE
bug: Firestore compositeFilter 'and' is not working 

### ISSUE THREAD

**Issue by aliexalter** [capawesome-team/capacitor-firebase#554]
### Plugin(s)

- [ ] Analytics
- [ ] App
- [ ] App Check
- [ ] Authentication
- [ ] Crashlytics
- [X] Cloud Firestore
- [ ] Cloud Messaging
- [ ] Cloud Storage
- [ ] Performance
- [ ] Remote Config

### Version

5.4.0

### Platform(s)

- [X] Android
- [X] iOS
- [ ] Web

### Current behavior

```
const monthData = await FirebaseFirestore.getCollection({
        reference: 'users/docs/entries',
        compositeFilter:{
          type: 'and',
          queryConstraints: [
            {
              type: 'where',
              fieldPath: 'Year',
              opStr: '==',
              value: format(this.currentDate, "yyyy")
            },
            {
              type: 'where',
              fieldPath: 'Month',
              opStr: '==',
              value: format(this.currentDate, "MM")
            }
          ]
        }
      });
```
When I change this.currentDate by adding year it returns same result for example
this.currentDate Year == 2024, Month == 02 return the 2 results suppose A,B
When I change to 
this.currentDate Year == 2023, Month == 02 it still return results A,B where as there is no record which has year 2023

### Expected behavior

It should return empty [] snapshot which it can return when only Month is changed

### Reproduction

Make a collection in Firebase Firestore with two fields and then query using compositeFilter

### Steps to reproduce

Make a collection in Firebase Firestore with two fields and then query using compositeFilter.

### Capacitor doctor

💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 5.7.0
  @capacitor/core: 5.7.0
  @capacitor/android: 5.7.0
  @capacitor/ios: 5.7.0

Installed Dependencies:

  @capacitor/cli: 6.0.0-rc.0
  @capacitor/core: 6.0.0-rc.0
  @capacitor/android: 6.0.0-rc.0
  @capacitor/ios: 6.0.0-rc.0

[success] iOS looking great! 👌
[success] Android looking great! 👌

### Before submitting

- [X] I have read and followed the [bug report guidelines](https://capawesome.io/contributing/bug-reports/).
- [X] I have attached links to possibly related issues and discussions.
- [X] I understand that incomplete issues (e.g. without reproduction) are closed.

**Comment by aliexalter** [comment]
Note: Year and Month Fields are strings in Firestore collection documents

**Comment by aliexalter** [comment]
I think problem is here where it uses FieldPath for Field too
![Screenshot 2024-02-14 at 3 14 27 PM](https://github.com/capawesome-team/capacitor-firebase/assets/2280503/85a757f5-843f-40d9-ab6f-324ef482a97c)
in above scenario
 fieldPath: 'Year' is actually a Field
 fieldPath: 'Month' is actually a Field
FieldPath looks for documentID and there is no document with id 'Year' or 'Month'

OR

Filter.and() is being used twice like Filter.and(Filter.and())

**Comment by robingenz** [comment]
I just tested this on Android and i can not reproduce the issue. Make sure that `format(this.currentDate, "yyyy")` and `format(this.currentDate, "MM")` outputs what you expect.

**Comment by aliexalter** [comment]
monthly.page.ts
```
import { Component, OnInit } from '@angular/core';
import { addMonths, addYears, format } from 'date-fns';
import { AuthenticationService } from 'src/app/services/authentication.service';
import { FirebaseFirestore } from '@capacitor-firebase/firestore';

Component({
  selector: 'app-monthly',
  templateUrl: './monthly.page.html',
  styleUrls: ['./monthly.page.scss'],
})
export class MonthlyPage implements OnInit {
public s_year:number = 0; //use for UI
  public s_month:string = ''; //use for UI
  private currentDate = new Date();
  public entries:any[] = [];

async ngOnInit() {
    this.s_year = this.currentDate.getFullYear();
    this.s_month = format(this.currentDate, 'MMMM');
  }

async ionViewWillEnter() {
    await this.loadCurrentMonthData();
  };

async prevYear() {
    this.currentDate = addYears(this.currentDate, -1);
this.s_year = this.currentDate.getFullYear();
    await this.loadCurrentMonthData();
  };
  async nextYear() {
    this.currentDate = addYears(this.currentDate, +1);
this.s_year = this.currentDate.getFullYear();
    await this.loadCurrentMonthData();
  };
  async prevMonth() {
    this.currentDate = addMonths(this.currentDate, -1);
this.s_month = format(this.currentDate, 'MMMM');
    await this.loadCurrentMonthData();
  };
  async nextMonth() {
    this.currentDate = addMonths(this.currentDate, +1);
this.s_month = format(this.currentDate, 'MMMM');
    await this.loadCurrentMonthData();
  };
  async loadCurrentMonthData() {
     console.log("loadCurrentMonthData", this.currentDate, format(this.currentDate, "yyyy"), format(this.currentDate, "MM"));
   this.entries = [];
      const monthData = await FirebaseFirestore.getCollection({
        reference: 'users/docs/entries',
        compositeFilter:{
          type: 'and',
          queryConstraints: [
            {
              type: 'where',
              fieldPath: 'Year',
              opStr: '==',
              value: format(this.currentDate, "yyyy")
            },
            {
              type: 'where',
              fieldPath: 'Month',
              opStr: '==',
              value: format(this.currentDate, "MM")
            }
          ]
        }
      });
      if(monthData.snapshots.length > 0)
      {
        monthData.snapshots.forEach((doc:any) => {
          let entry:any = doc.data;
          entry.id = doc.id;
          this.entries.push(entry);
        });
      }
  };
};
```

monthly.page.html
```
<ion-content [fullscreen]="true">
  <ion-grid>
    <ion-row class="yearChange">
      <ion-col>
        <ion-icon name="arrow-back" (click)="prevYear()"></ion-icon>
      </ion-col>
      <ion-col class="ion-text-center">
        <strong>{{s_year}}</strong>
      </ion-col>
      <ion-col>
        <ion-icon class="ion-float-right" name="arrow-forward" (click)="nextYear()"></ion-icon>
      </ion-col>
    </ion-row>
    <ion-row class="monthChange">
      <ion-col>
        <ion-icon name="arrow-back" (click)="prevMonth()"></ion-icon>
      </ion-col>
      <ion-col class="ion-text-center">
        <strong>{{s_month}}</strong>
      </ion-col>
      <ion-col>
        <ion-icon class="ion-float-right" name="arrow-forward" (click)="nextMonth()"></ion-icon>
      </ion-col>
    </ion-row>
</ion-grid>
</ion-content>
```

Here is the Firestore data screenshot
![Screenshot 2024-02-14 at 4 20 14 PM](https://github.com/capawesome-team/capacitor-firebase/assets/2280503/dd28d817-0d76-48b9-970d-70a3e567eb7c)

**Comment by aliexalter** [comment]
create at least 3 documents inside entries which have Fields with same Month:"02" and Year: "2024"

**Comment by robingenz** [comment]
Unfortunately, my time is limited and I can only work on issues that are sufficiently prepared.

As described above, please provide a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example) using [this template](https://github.com/capawesome-team/.capacitor-app) in a public GitHub repository so we can debug the issue.

Just add two buttons to your example:
1. Set document
2. Get document

**Comment by robingenz** [comment]
@aliexalter Please provide a **public** repository. Maybe someone else will have a similar problem in the future and take a look at this conversation and the code.

**Comment by aliexalter** [comment]
@robingenz its public now sorry for inconvenience

**Comment by robingenz** [comment]
I still can't reproduce your issue. Everything seems to be working as expected.

![grafik](https://github.com/capawesome-team/capacitor-firebase/assets/13857929/3a06019e-a472-45f1-93ba-9beb48fc17a5)

I also tested it with my own (empty) Firebase project:

![grafik](https://github.com/capawesome-team/capacitor-firebase/assets/13857929/9ff024ab-4cfe-495b-83d2-8b33c1c86022)

**Comment by aliexalter** [comment]
Have you try changing the Year value in composite query? it should not return 2025 or 2023 result it will be empty. But still return the results.

**Comment by aliexalter** [comment]
Check the log when changing year by tapping prev and next buttons

**Comment by robingenz** [comment]
Thank you for pointing that out. I've just created a fix: #562
